code/game/objects/structures/curtains.dm

```diff
@@ -30,6 +32,49 @@
 	if(istype(user, /mob/living/silicon/robot) && Adjacent(user)) // Robots can open/close it, but not the AI.
 		attack_hand(user)
 
+/obj/structure/curtain/attackby(obj/item/W, mob/user)
+
+	if(W.iswirecutter())
+		if(manipulating)	return
+		manipulating = TRUE
+		visible_message(span("notice", "[user] begins cutting down \the [src]."),
+					span("notice", "You begin cutting down \the [src]."))
+		if(!do_after(user, 30/W.toolspeed))
+			manipulating = FALSE
+			return
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		visible_message(span("notice", "[user] cuts down \the [src]."),
+					span("notice", "You cut down \the [src]."))
+		if(istype(src, /obj/structure/curtain/open/medical))
+			new /obj/item/stack/material/plastic(src.loc)
+		else
+			new /obj/item/stack/material/cloth(src.loc, 2)
+		qdel(src)
+	else if(W.sharp && !W.noslice)
+		if(manipulating)	return
+		manipulating = TRUE
+		visible_message(span("notice", "[user] begins cutting down \the [src]."),
+					span("notice", "You begin cutting down \the [src]."))
+		if(!do_after(user, 30))
+			manipulating = FALSE
+			return
+		playsound(src.loc, pick('sound/items/rip1.ogg','sound/items/rip2.ogg'), 50, 1)
+		visible_message(span("notice", "[user] cuts down \the [src]."),
+					span("notice", "You cut down \the [src]."))
+		if(!istype(src, /obj/structure/curtain/open/medical))
+			new /obj/item/stack/material/cloth(src.loc, 1)
+		qdel(src)
```

Comment:
This entire block is duplicate code and can definitely be condensed into the former one